style_guides/architecture.md

```diff
@@ -0,0 +1,23 @@
+# Architecture Style Guide
+
+## File and Folder Structure
+
+Kibana applications, both the core project and any plugins created for Kibana, should follow this basic file and folder structure:
+
+```
+plugin_root:
+
+.
+├── public/
+├── server/
+├── common/
+└── index.js
+```
+
+The **public** folder is where you place client-side code for your application. This folder will automatically be mounted at a prefix starting with `plugins` and your application's `id`. For example, if your application's `id` is *utile*, and you were trying to import a module from `public/lib/formatter.js`, you would import modules as `plugins/utile/lib/formatter`.
```

Comment:
Ah, I follow you. As a plugin developer, it's important to understand how this works though. It's how we reference public code from the server, in `uiExports` and `hacks`, for example. It's not a pattern that's going to be easy to do away with, so I don't see a reason to *discourage* it here.

Maybe just mentioning it as a NOTE, kind of like an alternative, so it looks less like we're encouraging it is enough?